server/src/test/java/org/elasticsearch/index/mapper/DocumentParserTests.java

```diff
@@ -59,20 +60,50 @@
         return List.of(new DocumentParserTestsPlugin(), new TestRuntimeField.Plugin());
     }
 
-    public void testDynamicUpdateWithRuntimeField() throws Exception {
+    public void testParseWithRuntimeField() throws Exception {
         DocumentMapper mapper = createDocumentMapper(runtimeFieldMapping(b -> b.field("type", "test")));
-        ParsedDocument doc = mapper.parse(source(b -> b.field("test", "value")));
-        RootObjectMapper root = doc.dynamicMappingsUpdate().root;
-        assertEquals(0, root.runtimeFieldTypes().size());
-        assertNotNull(root.getMapper("test"));
+        ParsedDocument doc = mapper.parse(source(b -> b.field("field", "value")));
+        //field defined as runtime field but not under properties: no dynamic updates, the field does not get indexed
+        assertNull(doc.dynamicMappingsUpdate());
+        assertNull(doc.rootDoc().getField("field"));
     }
 
-    public void testDynamicUpdateWithRuntimeFieldSameName() throws Exception {
-        DocumentMapper mapper = createDocumentMapper(runtimeFieldMapping(b -> b.field("type", "test")));
+    public void testParseWithShadowedField() throws Exception {
+        XContentBuilder builder = XContentFactory.jsonBuilder()
+            .startObject().startObject("_doc")
+                .startObject("runtime")
+                  .startObject("field").field("type", "test").endObject()
+                .endObject()
+                .startObject("properties")
+                  .startObject("field").field("type", "keyword").endObject()
+                .endObject()
+            .endObject().endObject();
```

Comment:
When we hit this file with the formatter one day it'll make this look like garbage. Would you be ok running the formatter on this now and playing with the output some to make it look ok when the formatter runs?

Reply:
You can do something like:
```
createDocumentMapper(topMapping(b -> {
    b.startObject("runtime");
    {
        b.startObject("field").field("type", "test").endObject();
    }
    b.endObject();
    b.startObject("properties");
    {
        b.startObject("field").field("type", "keyword").endObject();
    }
    b.endObject();
});
```

Keeps the formatter happy and there's a bit less ceremony around builders.